knowledge-base/dockmanager-reset-state.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Reset DockManager State on Button Click in Blazor
+description: Learn how to reset the DockManager state in Blazor using a button click and save the default state after the initial render.
+type: how-to
+page_title: How to Reset DockManager State Dynamically in Blazor
+meta_title: Reset DockManager State Dynamically in Blazor
+slug: dockmanager-kb-reset-state
+tags: dockmanager, blazor, state, reset
+res_type: kb
+ticketid: 1691957
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>DockManager for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+## Description
+
+I want to reset the state of the [DockManager](slug:dockmanager-overview) component on a button click. The DockManager currently only resets by reloading the page. I need a solution to reset its state dynamically.
+
+This knowledge base article also answers the following questions:
+- How to reset DockManager layout to its default state?
+- How to refresh DockManager without reloading the page?
+- How to implement a button to reset DockManager panes?
+
+## Solution
+
+To reset the DockManager state dynamically on a button click:
+
+1. Capture the Default State After Initial Render&mdash;Save the default state when the DockManager is first rendered using the [`OnAfterRenderAsync`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.componentbase.onafterrenderasync?view=aspnetcore-9.0) lifecycle method
```

Comment:
This step contains two duplicate statements, one of which is less specific and actionable. Consider shortening.

```suggestion
1. Capture and save the default DockManager state in the [`OnAfterRenderAsync`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.componentbase.onafterrenderasync?view=aspnetcore-9.0) lifecycle method.
```